### PR TITLE
Add firebase push notification plugin

### PR DESCRIPTION
**Pull request by jamshale**
Adds the firebase push notification plugin with documentation, unit test and configurations.

****Note:**** Integration tests not completed. Quite involved, as it requires setting up a mediated connection and a dummy webhook server to make sure events are being triggered and attempting to contact firebase server. Will create another ticket for this task.

### CODE REVIEW

File: firebase_push_notifications/firebase_push_notifications/v1_0/manager.py
Comment (amanji) [inline firebase_push_notifications/firebase_push_notifications/v1_0/manager.py:31]: Small nitpick but usually the linter complains if there's no space between the docstring and the first line of code. I'm guessing that if you mark the line with `# noqa` the linter doesn't complain

**Review by amanji — COMMENTED** [review]
LGTM so far. I've added some very minor comments that I leave to your discretion whether they need to be addressed or not.

File: firebase_push_notifications/firebase_push_notifications/v1_0/manager.py
Comment (esune) [inline firebase_push_notifications/firebase_push_notifications/v1_0/manager.py:54]: Should this be `logger.error` instead?

File: firebase_push_notifications/firebase_push_notifications/v1_0/manager.py
Comment (esune) [inline firebase_push_notifications/firebase_push_notifications/v1_0/manager.py:71]: Depending on how "chatty" we want the mediator service to be, statements like this might be better as `debug` messages.

**Review by esune — COMMENTED** [review]
Code looks good overall. Just a comment about log levels: with high load printing a high volume of statements to console might cause performance issues, and make the service very verbose - unsure on whether we want that by design or not.

File: firebase_push_notifications/firebase_push_notifications/v1_0/routes.py
Comment (jamshale) [reply firebase_push_notifications/firebase_push_notifications/v1_0/routes.py:125]: This was just matching the rfcs and the corresponding credo library.
https://github.com/hyperledger/aries-rfcs/blob/main/features/0734-push-notifications-fcm/README.md

File: firebase_push_notifications/README.md
Comment (jamshale) [reply firebase_push_notifications/README.md]: Thanks. Ya I copied this from hackmd and thought it was the same.

**Comment by jamshale** [comment]
Updated to address comments and change logging.

**Review by esune — APPROVED** [review]
LGTM 👍🏻

**Comment by jamshale** [comment]
@amanji Could you approve this or update any comments. I need a maintainer approval to merge.

File: firebase_push_notifications/firebase_push_notifications/v1_0/handlers/handler.py
Comment (amanji) [inline firebase_push_notifications/firebase_push_notifications/v1_0/handlers/handler.py:21]: Not sure if you need to run the linter/formatter

File: firebase_push_notifications/firebase_push_notifications/v1_0/routes.py
Comment (amanji) [reply firebase_push_notifications/firebase_push_notifications/v1_0/routes.py:125]: That makes sense for the DIDComm URL, but the admin endpoint deviates in structure from the rest of the admin endpoints.

File: firebase_push_notifications/firebase_push_notifications/v1_0/handlers/handler.py
Comment (jamshale) [reply firebase_push_notifications/firebase_push_notifications/v1_0/handlers/handler.py:21]: The linter is run on all PR's and it's passing... So it should be fine.

File: firebase_push_notifications/firebase_push_notifications/v1_0/routes.py
Comment (jamshale) [reply firebase_push_notifications/firebase_push_notifications/v1_0/routes.py:125]: Oh. I see what you mean. I think I was just trying to match the DIDComm URL. I can probably just remove the `1.0` then.

File: firebase_push_notifications/firebase_push_notifications/v1_0/handlers/handler.py
Comment (amanji) [reply firebase_push_notifications/firebase_push_notifications/v1_0/handlers/handler.py:21]: Yeah, resolved this after I wrote it. Not an issue.

File: firebase_push_notifications/firebase_push_notifications/v1_0/routes.py
Comment (amanji) [reply firebase_push_notifications/firebase_push_notifications/v1_0/routes.py:125]: I know it's super nitpicky

**Review by amanji — APPROVED** [review]
LGTM so far!

**Review by amanji — APPROVED** [review]

File: firebase_push_notifications/firebase_push_notifications/v1_0/routes.py
Comment (jamshale) [reply firebase_push_notifications/firebase_push_notifications/v1_0/routes.py]: No. That's a good catch. I didn't notice. Did this a long time ago.

Comment (jamshale) [reply firebase_push_notifications/firebase_push_notifications/v1_0/routes.py]: I updated it.

**Pull request merged**
